napari/_qt/widgets/qt_progress_bar.py

```diff
@@ -44,6 +46,11 @@ def __init__(
 
         self.setLayout(base_layout)
 
+        if isinstance(prog, cancelable_progress):
+            self.cancel_button = QPushButton(trans._('Cancel'))
+            self.cancel_button.clicked.connect(self._cancel)
+            pbar_layout.addWidget(self.cancel_button)
+
```

Comment:
Conditional adding attribute is not the best practice. Maybe `self.cacnel_button.setVisible(isinstance(prog, cancelable_progress))`

Reply:
Added it in cb34bc93